src/test/java/com/marklogic/kafka/connect/source/ConstraintStorageTest.java

```diff
@@ -0,0 +1,87 @@
+            MarkLogicSourceConfig.CONSTRAINT_STORAGE_COLLECTIONS, constraintStorageCollections
+        );
+        List<SourceRecord> newRecords = task.poll();
+        Assertions.assertEquals(15, newRecords.size(), "All 15 initial records should be returned");
+        String previousMaxConstraintColumnValue = task.getPreviousMaxConstraintColumnValue();
+
+
+        ConstraintValueStore constraintValueStore = new MarkLogicConstraintValueStore(getDatabaseClient(), constraintStorageUri,
+            constraintColumnName, new HashMap<>());
+        String lastValue = constraintValueStore.retrievePreviousMaxConstraintColumnValue();
+        Assertions.assertEquals(previousMaxConstraintColumnValue, lastValue,
+            "The stored lastValue should equal the value from the Task");
+        PermissionsTester permTester = this.readDocumentPermissions(constraintStorageUri);
+        permTester.assertReadPermissionExists(roleName);
+        permTester.assertUpdatePermissionExists(roleName);
+        this.assertInCollections(constraintStorageUri, constraintStorageCollections.split(","));
+
+        String laterTime = "02:01:00";
+        loadSingleAuthorRowIntoMarkLogicWithCustomTime("firstHigherId", "6", laterTime, "firstHigherId");
+        loadSingleAuthorRowIntoMarkLogicWithCustomTime("secondHigherId", "7", laterTime, "secondHigherId");
+        loadSingleAuthorRowIntoMarkLogicWithCustomTime("lowestId", "0", laterTime, "lowestId");
+
+        newRecords = task.poll();
+        Assertions.assertEquals(2, newRecords.size(), "Only 2 of the 3 new records should be returned");
+        Assertions.assertTrue(newRecords.toString().contains("\"Medical.Authors.ID\":{\"type\":\"xs:integer\",\"value\":6}"));
+        Assertions.assertTrue(newRecords.toString().contains("\"Medical.Authors.ID\":{\"type\":\"xs:integer\",\"value\":7}"));
+
+
+        String newLastValue = constraintValueStore.retrievePreviousMaxConstraintColumnValue();
+        previousMaxConstraintColumnValue = task.getPreviousMaxConstraintColumnValue();
+        Assertions.assertEquals(previousMaxConstraintColumnValue, newLastValue,
+            "The stored lastValue should have been updated to the value from the second run of the task");
+
+
+        String evenLaterTime = "03:01:00";
+        loadSingleAuthorRowIntoMarkLogicWithCustomTime("thirdHigherId", "8", evenLaterTime, "thirdHigherId");
+        loadSingleAuthorRowIntoMarkLogicWithCustomTime("fourthHigherId", "9", evenLaterTime, "fourthHigherId");
+
+        storeBadConstraintState(constraintStorageUri);
+        String badLastValue = constraintValueStore.retrievePreviousMaxConstraintColumnValue();
+        Assertions.assertNull(badLastValue, "The retrieved last value should be null since the document is invalid");
+
+        newRecords = task.poll();
+        Assertions.assertEquals(20, newRecords.size(),
+            "Since the stored state document is invalid, all 21 records should have been returned");
```

Comment:
Typo, should be "20" - though I would just do "all" to avoid the chance of the number being incorrect